File: knowledge/src/main/java/com/ayvytr/knowledge/model/GankHistoryContentModel.java
package com.ayvytr.knowledge.model;

import com.ayvytr.commonlibrary.bean.GankHistoryContent;
import com.ayvytr.commonlibrary.server.GankApi;
import com.ayvytr.knowledge.contract.GankHistoryContentContract;
import com.ayvytr.network.ApiClient;

import io.reactivex.Observable;

/**
 * @author admin
 */
public class GankHistoryContentModel implements GankHistoryContentContract.Model {
    private GankApi mGankApi;

    public GankHistoryContentModel() {
        mGankApi = ApiClient.getInstance().create(GankApi.class);
    }

    @Override
    public void onDestroy() {
        mGankApi = null;
    }

    @Override
    public Observable<GankHistoryContent> getGankByDate(String year, String month, String dayOfMonth) {
        return mGankApi.getDataByDate(year, month, dayOfMonth);
    }
}

File: knowledge/src/main/java/com/ayvytr/knowledge/model/GankHistoryContentModel.kt
package com.ayvytr.knowledge.model

import com.ayvytr.commonlibrary.bean.GankHistoryContent
import com.ayvytr.commonlibrary.server.GankApi
import com.ayvytr.knowledge.contract.GankHistoryContentContract
import com.ayvytr.network.ApiClient

import io.reactivex.Observable

/**
 * @author admin
 */
class GankHistoryContentModel : GankHistoryContentContract.Model {
    private var mGankApi: GankApi? = null

    init {
        mGankApi = ApiClient.getInstance().create(GankApi::class.java)
    }

    override fun onDestroy() {
        mGankApi = null
    }

    override fun getGankByDate(year: String, month: String, dayOfMonth: String): Observable<GankHistoryContent> {
        return mGankApi!!.getDataByDate(year, month, dayOfMonth)
    }
}
